lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/GenericMathLangAST.hs

```diff
@@ -46,21 +46,21 @@ import Language.Haskell.TH.Syntax (Lift)
 
 -- | GML types that are declared in L4 by the user, e.g. 'Person' or 'Singaporean citizen'.
 --
-data L4EntType =
-    L4EntType T.Text  -- ^ a user-defined named type
+data L4EntityType =
+    L4EntityType T.Text  -- ^ a user-defined named type
   | L4Enum [T.Text]   -- ^ a user-defined enumeration type
-  | L4List L4EntType  -- ^ a user-defined list type
-  deriving stock (Eq, Generic, Show)
+  | L4List L4EntityType  -- ^ a user-defined list type
+  deriving stock (Eq, Generic, Show, Ord)
   deriving anyclass (Hashable)
 
 -- | Turn a surface-L4 type signature into a GML type.
 --
 -- NOTE / TODO: This translation seems lossy.
 --
-mkEntType :: TypeSig -> L4EntType
+mkEntType :: TypeSig -> L4EntityType
```

Comment:
Nice! Yes, `L4EntityType` is a lot more understandable than `L4EntType`. Thank you for changing it.